source/common/upstream/cluster_manager_impl.h

```diff
@@ -260,6 +263,7 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
     Event::Dispatcher& thread_local_dispatcher_;
     std::unordered_map<std::string, ClusterEntryPtr> thread_local_clusters_;
     std::unordered_map<HostConstSharedPtr, ConnPoolsContainer> host_http_conn_pool_map_;
+    std::unordered_set<Envoy::Upstream::ClusterUpdateCallbacks*> update_callbacks_;
```

Comment:
I think ordering this might make sense for debugging; otherwise we get non-deterministic callback ordering.

Reply:
+1 I would just use a vector probably given low rate of updates.